매우 기본적인? 피드 구현

somnisomni opened the pull request:
피드 아이템을 좌롸락 나열할 수 있는 피드 페이지의 기본적인 구현입니다.

피드 부분 외에도 앱 전반적으로 업데이트된 로직이 있습니다.

somnisomni committed:
Ignore annoying `max-len` ESLint rule

섣불리 개행을 넣으면 whitespace 관련 문제가 생길 수 있는 HTML 코드에서도
Airbnb 프리셋에 따라 한 줄에 100자 넘지 말라고 해서
그냥 해당 ESLint 규칙을 비활성화합니다.

somnisomni committed:
More styling work for Feed item

또한 내용의 길이가 길어 overflow가 생길 경우 max-height에 따라 내용을 자릅니다.
overflow 생길 때 자른 부분을 fade-out 하는 스타일은 나중에 구현할 예정입니다.

somnisomni committed:
Simulate "Likes" process

somnisomni committed:
Navigate to article page when click Feed item

somnisomni committed:
Shorten IFeedItem code

somnisomni committed:
Use template literals for router path

somnisomni committed:
Auto-update upload date formatting in Feed item

somnisomni committed:
Add elevation to profile image in Feed item

wooksong commented on Web/src/pages/feed/MainFeedPage.vue:
```diff
@@ -1,13 +1,73 @@
 export default class MainFeedPage extends Vue {
-
+  get feedItems(): Array<IFeedItem> {
+    return [
```
테스트 또는 더미 데이터의 용도시면, 이러한 하드 코딩/정적인 데이터보다는 단위 테스트를 넣어서 빌드 후에, 동적으로 데이터를 넣는다든지 (넣는 것은 runtime에 넣는 것이지만 데이터는 미리 만들어져야 하죠. 재현성을 위해), 아니면 옵션을 받아서 (i.e., 메인 이미지가 있고 없고, 작성 시간의 분포 등등) 빌드 타임에 생성한다든지 (이건 재현성을 위해서는 빌드가 성공하면 해당 데이터를 덤프하거나 로그를 남기는 기능이 필요하겠죠) 등의 방법이 더 좋을 것 같습니다.

근데 시간이 많지 않으니 나중에 심심하면 고민해보셔요.

somnisomni replied:
리뷰 감사합니다! 저도 테스트데이터를 커밋에 올리는건 좋은 principle이 아니란건 대충 짐작하고는 있었는데, 커밋 당시에는 테스트로라도 call해볼 수 있는 백엔드가 세팅되어있지 않았던 것도 있고 솔직히 말하면 그 외 어떤 방법으로 하드코딩 안하고 테스트데이터를 올릴 수 있을지도 잘 몰라서 이런 식으로 되어버린 것 같습니다...

참고하여 더 공부해보겠습니다!

somnisomni committed:
Implement basic login form

somnisomni committed:
Workaround for `.row` negative margin

`v-layout`는 실제 렌더링 시 `.row` 클래스가 붙는데,
`.row` 클래스에는 negative margin이 적용되어 있습니다. 이로 인해
피드 아이템의 메인 이미지 유무에 따라 아이템 간 마진이 일정
간격으로 정확히 적용되지 않는 문제가 발생하였습니다.

top margin을 리셋(0px로 설정)하는 것으로 workaround합니다.

somnisomni committed:
Large screen style work for Feed page

somnisomni committed:
Overflow style work for Feed items

+ 추가로 댓글/좋아요 아이콘과 숫자 사이에 마진을 약간 추가합니다.

somnisomni committed:
Add Feed/Discover/Pool tabs in navbar

somnisomni committed:
Add horizontal padding to Feed item reaction btns

somnisomni committed:
Lower Feed page's max-width

somnisomni committed:
Implement bottom tab navigation

작은 화면에서 피드/발견/풀 탭 네비게이션이 추가된 하단 네비바에도 표시됩니다.

해당 탭 네비게이션을 하단 네비바에도 재사용하기 위해 `AppTabNavigation.vue`로
컴포넌트화하는 작업이 이루어졌습니다.

큰 화면에서 상단 네비바에 표시되는 탭과 작은 화면에서 하단 네비바에 표시되는 탭은
동일한 내용을 갖습니다.

somnisomni committed:
Categorize imports with comments in `main.ts`

somnisomni committed:
Implement basic Feed item

somnisomni committed:
Disable page horizontal overflow

somnisomni committed:
Layout reaction area of feed item

+ `IFeedItem`에 댓글/좋아요 수 속성 추가

somnisomni committed:
Wrap `router-view` with `v-container`

Vuetify 문서에서 제시된 예시를 따릅니다.
이제 콘텐츠의 스타일이 Vuetify가 의도한 대로 표시됩니다.
참고 : https://vuetifyjs.com/en/components/application/

somnisomni committed:
Harden(?) example feed data

somnisomni committed:
Format upload date with "... time ago"

timeago.js(https://www.npmjs.com/package/timeago.js/v/4.0.0-beta.3)를 이용하여
피드 아이템 내 업로드 일자를 포맷합니다.

somnisomni commented:
※ 참고: git rebase 하다가 일부 커밋의 커밋 시간/일자(부모-자식 관계)가 꼬여버렸습니다... :cry: :sob:

somnisomni committed:
Use Vuex store for login state

somnisomni committed:
Componentize Account menus in App bar

somnisomni committed:
Lint on `.eslintrc.js` (lol)

somnisomni committed:
Set `required` opt for login/logout callback prop

somnisomni committed:
Mark main image url is optional in IPostDisplay

somnisomni committed:
Use `IPostDisplay` type in `IFeedItem`
